### ISSUE TITLE
Parser breaking when using the same attributes as Tap class methods

### ISSUE THREAD

**Issue by mrapacz** [swansonk14/typed-argument-parser#18]
Hi, first of all thanks a lot for working on this Python package, it's great! 🎉 

I wanted to let you know about a potential issue I've stumbled into recently. 
To give you some context - I was working on a Python package which generates code for NamedTuple classes based on e.g. json files. Now I wanted to give the CLI user an option to specify a boolean `--as-dict` flag, which, if set, would add nice `as_dict()` methods to the generated classes. However, it turned out that giving such a name to the flag was a mistake.

To give you a minimal example to reproduce this:
```python
from tap import Tap

class MyParser(Tap):
	as_something: bool = True

if __name__ == '__main__':
	args = MyParser().parse_args()
	print(args)
```
If I save this as `main.py` and try to run it, I'll get:
```bash
$ python main.py
Traceback (most recent call last):
  File "main.py", line 8, in <module>
    print(args)
  File "/Users/mrapacz/opt/anaconda3/envs/tap_bug/lib/python3.8/site-packages/tap/tap.py", line 412, in __str__
    return pformat(self.as_dict())
TypeError: 'bool' object is not callable
```
At the same time, though, if I rename the flag to something else, the error doesn't appear:
```python
from tap import Tap

class MyParser(Tap):
	as_whatever: bool = True

if __name__ == '__main__':
	args = MyParser().parse_args()
	print(args)
```
```bash
$ python main.py
{'as_whatever': True}
```
The issue seems to be a result of naming collision between the flag and the method in the Tap class. When Python wants to call `self.as_dict()`, if turns out it's not callable (since at that point it's no longer a callable, but a boolean). 

The setup I'm using:
```bash
$ python --version
Python 3.8.2
$ pip freeze
certifi==2019.11.28
mypy-extensions==0.4.3
typed-argument-parser==1.4.1
typing-extensions==3.7.4.2
typing-inspect==0.5.0
```

I'm not sure if this is something you'd like to fix, but if not, I guess it would be valuable to mention somewhere, so as to save some debugging time from other developers 😅 
Thanks a lot again!

**Comment by martinjm97** [comment]
Hi, glad to hear your enjoying Tap!

I think you meant 
```
class MyParser(Tap):
	as_dict: bool = True
```
instead of
```
class MyParser(Tap):
	as_something: bool = True
```
in your example. I reproduced your using the modified version. 

If I understand your issue correctly, this is already marked as [an issue](https://github.com/swansonk14/typed-argument-parser/issues/12). Or am I mistaken?

We should guard against shadowing in this way, raising an error. Sorry for the lost time spent debugging and we'll try to fix this bug soon!

P.S. `ntgen` is cool!
P.P.S. I added to the issue linked to include the example and to be more clear, so don't be surprised if it looks similar :).

**Comment by mrapacz** [comment]
Damn, yes, you're right concerning both the typo in the example and the fact that I duplicated the already existing issue! I'm going to close this one, then. Thanks a lot for the response and sorry for not going through the existing issues thoroughly in the first place 🤦‍♂ 

P.S About ntgen, thanks a lot, glad to hear that! I've just started working on it recently, hope it becomes useful for at least a few members of the Python community 😅
